### ISSUE TITLE
User unable to upload data - project 48

### ISSUE THREAD

**Issue by zcrysler** [MotusDev/Motus-TO-DO#277]
I don't have many details yet, but user 2600 was trying to upload data for project 48 to the new page, she said she selected verify, then upload, and then nothing happened. This happened on June 15 and 18th.

She also said that users 2510 and 2640 were denied access to the page, both those users have update permission for data, but are set to read only for project: https://motus.org/data/project/users?projectID=48. Could that be why or is it unrelated?

All users are in the field/on the road so response time will be slow. What specific information is needed from them to narrow down the problem?

**Comment by pmorrill** [comment]
It's not clear to me which step failed. When she selected 'verify' - did the file verify? Did she see the message that it is a new file and can be uploaded to Motus?

The first time she did this we might not have updated the live server with the fix we worked out (#263). The 2nd time it should have worked, so we are possibly looking at a different bug. She should retest, ignoring her earlier problem (ie, on the 15th) and tell us precisely what happens. If it still fails we need to see the file.

I will look into the permission issue for the other two users.

**Comment by zcrysler** [comment]
She was able to verify, and it completed properly. But then when she clicked complete upload nothing happened. I've asked her to retry and if it fails she'll send me the file and I'll pass it on to you

I just got a similar email from project 147:
"I keep trying to upload tower detection data (Tahoka 06182018.zip, specifically). It gets stuck at 32.1% every time. It says 'a partial copy of this file exists...' but trying to restart or resume does nothing." I've asked her to send me the file as well, let me know what other info you need

**Comment by zcrysler** [comment]
I tried uploading logged in as Kristen again, this is what shows up with the debugger, after verification, when I click either resume or upload
![screen shot 2018-06-26 at 11 44 15 am](https://user-images.githubusercontent.com/22080197/41932627-4eb6d9c8-7936-11e8-8c7a-2aec834a58ec.png)

**Comment by zcrysler** [comment]
Since I had started one logged in as myself, now when I try to resume from that file I run into the same issue as when logged in as Kristen

**Comment by pmorrill** [comment]
Yes - I have seen that now to. It takes awhile for me to get that far with my slow connection! There is an issue with the 'resume' option, definitely. I am tracking it right now.

Can you try choosing 'restart option' and see what you see?

**Comment by zcrysler** [comment]
The same thing happens, setting to restart and clicking upload doesn't do anything and results in another line in the console "SyntaxError: JSON.parse: unexpected character at line 3 column 1 of the JSON data[Learn More]"

**Comment by pmorrill** [comment]
ok - thanks. That saves me testing it. At least we know the issue. I will get it sorted today.

**Comment by pmorrill** [comment]
I have also edited the permission check in the servlet, so that anyone with 'detections data edit' privilege should be able to use it. Note that the servlet also allows tag registration files to be uploaded. Do we need to be more granular in the permissions check, when files come in? That would be a little more involved.

**Comment by zcrysler** [comment]
I tried uploading a file logged in as user 2600 in beta: I verified the file and it said "A partial copy of this file exists on the Motus server, possibly because of a network connection error when you last attempted to process the file. If you have made absolutely no changes to the file, you may resume uploading the file from where it left off. Otherwise, select the option to restart the upload. ", I selected restart upload and clickd "upload file for processing". It appeared to have finished upload and said "Your file is being placed in the processing queue.", but nothing's showing up on the status page for project 48, does beta not actually process the data?

I'm able to get to the upload page as the other users on beta now though so that issues been resovled

**Comment by pmorrill** [comment]
The data is processed on the SG server, so whether you upload by beta or live (or even sandbox) does not matter. Maybe the queue is just slow to process the file. Maybe Jermey has a way of gauging that.

**Comment by zcrysler** [comment]
Usually it shows up on the status page right away, if it's in the queue it will just say it's waiting to be processed. It still hasn't shown up

**Comment by Hussell** [comment]
The most recent file on sgdata.motus.org which was uploaded by user 2600 was uploaded on June 14th.

**Comment by zcrysler** [comment]
I tried uploading again, same result. If I click "restart", it shows the upload percentage increasing to 100, then says it's in the queue. If I click resume, it immediately says it's in the queue, but neither attempts seem to get the data in the queue

**Comment by pmorrill** [comment]
The file must be on the tmp folder, but not being moved to the main folder where the SG server picks it up. I'll try it from here.

I think that we need an admin tool that allows admin web-users to see what is on the two main folders that we use here. File names and sizes.

**Comment by pmorrill** [comment]
Quick fix is available: rename the file to

6.14.2018_marshland1SGdata1B-20180615T154836Z-001.zip

(The presence of the hash - # - character in the original file name is not being handled by the javascript that sets up the ajax calls. Remove that # character and the file looks to process in my local sandbox)

**Comment by denislepage** [comment]
Live has been pushed and restarted

**Comment by pmorrill** [comment]
The jscript encodeURIComponent() function seems to handle the hash character and allow us to send it as part of the file name. ie:

https://www.w3schools.com/jsref/jsref_encodeuricomponent.asp

**Comment by zcrysler** [comment]
So now that the other users can upload to project 48, there's a new issue. I logged in as user 2640, verified the file, got the message that it was new and I could upload. When I clicked "upload file for processing" nothing happens. I turned on debugging and got this:

SyntaxError: JSON.parse: unexpected character at line 2 column 1 of the JSON data[Learn More]

The file was called 2018-6-21 SGdata Bucktoe.7z, Paul I can send it to you if you'd like

**Comment by pmorrill** [comment]
Yes - better transfer it to me when you can, I'll give it a try.

**Comment by pmorrill** [comment]
Is this file also for upload to project 48?

**Comment by zcrysler** [comment]
yes, and the user is now reporting that the warning that he's not authorized to edit the project came back

**Comment by pmorrill** [comment]
Yes - it looks like to be a permission issue. I will get it sorted.

**Comment by zcrysler** [comment]
user 2572 is also having issues uploading to project 162, she says that the file verifies fully but when she clicks "upload for processing" nothing happens, even if she waits quite a while. She says if she clicks it a second time it says it's putting the file in the queue and then quickly follows up with "your file has been corrupted during upload, please try again". She said she's tried multiple times on safari and chrome with the same results. I got the same warning about a corrupt file earlier this week, I thought it was my slow internet but since she's having the same thing pop up repeatedly maybe it's something else. I've got the file and I can open it fine so the file itself isn't corrupt

**Comment by pmorrill** [comment]
I've just put in a fix for the permission problem. That was my bad as I skipped i nwhen making the original permission adjustment last week. The user you describe above (2572 on 162) is in a similar boat: data update permissions but no project update permissions. Therefore she would be having problems, no question.

Let's get this fix into beta on next deployment (Tuesday) and test both of this situation, and the previous one (user 2640 on 48).

**Comment by zcrysler** [comment]
When logged into beta as user 2572, I can't get to any of the manage data pages to test this, I get:

The page isn’t redirecting properly

An error occurred during a connection to beta.motus.org.

    This problem can sometimes be caused by disabling or refusing to accept cookies.

I'm on firefox and am accepting cookies. When logged in as myself on beta I'm able to navigate fine

**Comment by pmorrill** [comment]
I checked her permissions: the tool says she has NO permissions at the project level (but update on other levels). Does she not at least need read permission on the project just to get to the main landing page?

**Comment by Hussell** [comment]
Looks like if you have manage_project = 0 on all projects you're registered with the /data/ page goes into an infinite redirect loop, similar to what used to happen to users who weren't registered with any project.

**Comment by zcrysler** [comment]
Upload on beta for user 2640 on project 48 worked. So just need to get user 2572 going

**Comment by pmorrill** [comment]
Zoe: go into the project 'Manage users' and set the project access level to 'READ' for user 2572, then re-test.
Jeremy is right that there is a problem, but it is with the way the redirect is handled, which can be fixed independently.

Just to be clear: I believe that project READ is required for users to access the landing page, in order to get to more advanced tools.

The user might also get to a tool directly via the URL (skipping the landing page), but that is not practical, if possible at all.

Why do we even allow Project access permission to be set to NONE for project_users? Seems more complicated than it needs to be!

**Comment by Hussell** [comment]
With the access request tool I've been working on, I've been asked to set the permissions to 0 for manage_project and 1 for all the others. The intent is to allow the user to download data but not see things like landowner info.

**Comment by pmorrill** [comment]
Jeez - could we think of more complications? Alright: so what permission do people need to just see the landing page? Just be linked to the project via projects_users?

**Comment by zcrysler** [comment]
So prior to updating her permissions: when logged in as user 2572, I was able to get to the various data management pages by going there directly from the drop down menu which explains why she was able to upload in the first place (I was originally trying from the landing page). I tried uploading and had the same issue where after verification completed, clicking upload did nothing.

I then changed her project access to read and tried again, now I can access the landing page and data upload went through correctly

**Comment by Hussell** [comment]
You can view the landing page without being registered with any projects - it's just empty except for "You must be linked to a Motus project to use the project Management tools. Please contact Motus staff for further information." Users who are registered with a project but don't have read permission should be treated exactly like users who are not registered with a project. (This is one of the many issues with users and permissions which it would be nice to make a coordinated effort to clean up.)

**Comment by zcrysler** [comment]
Except the way beta is currently running, if you're logged in and registered to a project, but project permission is set to none, you can't even get to the landing page. you get

The page isn’t redirecting properly

Firefox has detected that the server is redirecting the request for this address in a way that will never complete.

    This problem can sometimes be caused by disabling or refusing to accept cookies.

You can access the individual pages through the dropdown menu though

**Comment by pmorrill** [comment]
That's correct: Jeremy is pointing out an inconsistency in the current implementation, that leads to what you describe.

I will take a look at making it more consistent, tomorrow hopefully.

**Comment by pmorrill** [comment]
I have updated the UserLanding page so that it allows a user who does NOT have project READ access to still see the page, though the page will only display links appropriate to his/her level of access. There are still some inconsistencies in how permissions are checked in some of the tools: I am working on those separately, and will probably open a new issue for that. This latest change is ONLY affecting the set of links displayed on the userLanding page - it does not change any incorrect implementations. For example: the 'Register Tags' link will display for those who have 'TagEditor' access....but the link takes the user to the submit data page (upload data), which will reject the user if they don't also have DataEditor privilege.

I am working on these inconsistencies this morning.

**Comment by zcrysler** [comment]
The issue with users that do not have Read permissions accessing the user landing page is resolved. Unless there are other inconsistencies still to sort out this can be closed.
